subsys/net/lib/sockets/sockets.c

```diff
@@ -1457,8 +1457,7 @@ int zsock_poll_internal(struct zsock_pollfd *fds, int nfds, k_timeout_t timeout)
 		if (K_TIMEOUT_EQ(timeout, K_FOREVER)) {
 			poll_timeout = SYS_FOREVER_MS;
 		} else {
-			poll_timeout = k_ticks_to_ms_floor32(
-				sys_clock_timeout_end_calc(timeout));
+			poll_timeout = k_ticks_to_ms_floor32(timeout.ticks);
```

Comment:
Should`poll_timeout` be expressed as ticks? It seems that it should be a value in ms where anything < 0 is "infinity".

Reply:
I'm not sure I understand the comment; please see https://github.com/zephyrproject-rtos/zephyr/pull/36220/files where this function was changed from receiving ms to receiving ticks to support microseconds. But since `z_fdtable_call_ioctl` expects ms, it is here converted back to ms, including the case in which `timeout` is `K_FOREVER` then `poll_timeout` is set to `-1` like you said it should. Sorry if I misunderstood.

Reply:
Thanks for clarifying.